### PR TITLE
Fix various warnings

### PR DESCRIPTION
**Pull request by coreyfloyd**
Mostly about not sending non conforming object to event logging.
But also some nullability stuff, etc…

### CODE REVIEW

**Comment by montehurd** [comment]
Seems to have reduced the warnings I see. @coreyfloyd are you no longer seeing any warning? I see a couple still...

**Comment by coreyfloyd** [comment]
@montehurd thee are still a few

File: Wikipedia/Code/MWKList.h [hunk 1]
```diff
@@ -40,9 +40,9 @@ typedef id<NSCopying, NSObject> MWKListIndex;
 
 - (EntryType)entryAtIndex:(NSUInteger)index;
 
-- (EntryType __nullable)entryForListIndex:(IndexType)listIndex;
+- (EntryType __nullable)entryForListIndex:(MWKListIndex)listIndex;
```
Comment (bgerstle) [inline Wikipedia/Code/MWKList.h]: this effectively removes the generic aspect of the `MWKList` API, meaning all the entry-related APIs won't auto-complete to the type parameters for the specific subclass.  AFAICT, the warning here is in `MWKHistoryList.addEntry`, which should be using `entry.title`, not `entry.listIndex`.

File: Wikipedia/Code/WMFArticleContainerViewController.m [hunk 2]
```diff
@@ -228,7 +228,7 @@ - (WMFArticleListTableViewController*)readMoreListViewController {
     return _readMoreListViewController;
 }
 
-- (WMFArticleFetcher*)articleFetcher {
+- (nullable WMFArticleFetcher*)articleFetcher {
```
Comment (bgerstle) [inline Wikipedia/Code/WMFArticleContainerViewController.m]: why is this `nullable` if we always return nonnull?

**Comment by coreyfloyd** [comment]
@bgerstle updated

**Comment by bgerstle** [comment]
@coreyfloyd :+1: thanks! left a minor comment on the "append read more" section stuff in 352ab9f. what do you think about adding a guard & assertion there?

**Comment by bgerstle** [comment]
:+1:

**Pull request merged**